src/proxy/resolve.rs

```diff
@@ -58,13 +59,29 @@ pub struct MakeSvc<R, M> {
 
 /// Observes an `R`-typed resolution stream, using an `M`-typed endpoint stack to
 /// build a service for each endpoint.
-#[derive(Clone, Debug)]
-pub struct Discover<R, M> {
+pub struct Discover<R: Resolution, M: svc::Service<R::Endpoint>> {
     resolution: R,
     make: M,
+    makes: MakeStream<M::Future>,
```

Comment:
TIOLI: personally, I find `makes` a bit weird (and, I notice, just one typo away from the similarly-named `make`. WDYT of `make_futures` or `make_stream` instead? it's admittedly a bit longer.